### PR TITLE
S3 Polling Producer

### PR DESCRIPTION
**Pull request by skyqrose**
Asana Task: [🛠 Add Raw GPS to Delta](https://app.asana.com/0/616151179860796/1202728650102028/f)

The producer configuration will be:
```json
  "producers": {
     "raw_gps": {
       "type": "s3",
       "bucket": "mbta-ctd-trc-data",
       "path": "rtr/prod/LightRailRawGPS.json",
       "frequency": ???
     }
  }
```
Though I'm not sure what the frequency should be and how the consumer configuration is done.

This should be deployed to dev before merging to make sure I got the AWS configuration right in `prod.exs`.

Things I'll need you to do:
- [ ] Review
- [ ] Set the configuration in Secrets Manager
- [ ] Deploy to dev
- [ ] Deploy to prod
- [ ] Let me know where in s3 the stored data will be.

### CODE REVIEW

File: lib/delta/producer.ex [hunk 1]
```diff
@@ -11,7 +12,7 @@ defmodule Delta.Producer do
           {:url, binary}
           | {:frequency, non_neg_integer}
           | {:http_mod, module}
-          | {:filters, [filter]}
+          | {:filters, [Filter.t()]}
```
Comment (paulswartz) [inline lib/delta/producer.ex:15]: question: If this is moving to `Filter.t()` as the type, do you still need the `filter` type on line 16?

**Review by paulswartz — COMMENTED** [review]
The code looks good, but `delta-dev` doesn't have permission to read the relevant bucket. Can you do a Terraform PR to give `delta-dev` and `delta-prod` the relevant `GetObject` permission?

File: lib/delta/producer.ex (hunk 1 above)
Comment (skyqrose) [reply lib/delta/producer.ex:14]: fixed.

**Comment by skyqrose** [comment]
Terraform PR: https://github.com/mbta/devops/pull/547/checks

**Review by paulswartz — APPROVED** [review]
I'm seeing logs from Dev, so I think this is good to go.

**Pull request merged**